pkg/plugins/backendplugin/coreplugin/core_plugin.go

```diff
@@ -70,50 +71,50 @@ func (cp *corePlugin) Target() backendplugin.Target {
 }
 
 func (cp *corePlugin) CollectMetrics(_ context.Context, _ *backend.CollectMetricsRequest) (*backend.CollectMetricsResult, error) {
-	return nil, backendplugin.ErrMethodNotImplemented
+	return nil, plugins.ErrMethodNotImplemented
```

Comment:
Please use `.Errorf` here and below to create "actual  errors"

Reply:
I found that really difficult to use / not forget. That's why I changed pkg/api/response/response.go and accept a `errutil.Base` there.

If that's not enough or incorrect, I would rather:
 - Expose `plugins.Err*` as final `errutil.Error`s since those are meant to be used as they are?
 - If none of the above, at least expose a method in `errutil` so we can use `plugins.ErrMethodNotImplemented.Err()` rather than `plugins.ErrMethodNotImplemented.Errorf("%w", plugins.ErrMethodNotImplemented)`
 
 WDYT?